### PR TITLE
osd/scheduler: rely on copy ellision to move return val

### PR DESCRIPTION
**Pull request by tchaikov**
C++14 enforces copy ellision in this case.

also silences warning of

src/osd/scheduler/mClockScheduler.cc:122:21: warning: redundant move in return statement [-Wredundant-move]
  122 |     return std::move(ret);
      |            ~~~~~~~~~^~~~~

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/osd/scheduler/mClockScheduler.cc [hunk 1]
```diff
@@ -119,15 +119,15 @@ WorkItem mClockScheduler::dequeue()
   if (!immediate.empty()) {
     auto ret = std::move(immediate.back());
     immediate.pop_back();
-    return std::move(ret);
+    return ret;
```
Comment (sseshasa) [inline src/osd/scheduler/mClockScheduler.cc:122]: I recall having made this change after I observed opensuse and possibly one of the ubuntu builds (can't recall which) failing on this line after changing the return type to std::variant for the dequeue() function. Those specific builds probably need to be monitored. I can't provide details of the failures as shaman doesn't show older builds. Is there a way to retrieve details of older shaman builds?

Comment (tchaikov) [reply src/osd/scheduler/mClockScheduler.cc:122]: @sseshasa not sure how old ? jenkins does keep finite number / days of build logs though. if it's a reproducible build failure, why not reproduce it?

Comment (sseshasa) [reply src/osd/scheduler/mClockScheduler.cc:122]: @tchaikov Yes, the shaman build with these changes should reproduce it.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-osd-scheduler-move-kefu/6eb8685b765286a60f9488c00507875dc056afdf/

**Review by batrick — APPROVED** [review]
LGTM

**Review by sseshasa — APPROVED** [review]
LGTM

**Review by athanatos — APPROVED** [review]

**Pull request merged**